app/api/access_codes.py

```diff
@@ -96,6 +97,9 @@ class AccessCodeList(ResourceList):
     """
     List and create AccessCodes
     """
+    def before_post(self, args, kwargs, data):
```

Comment:
Nope, always try to use flask-rest-jsonapi methods over marshmallow's if available, and I don't know what do you mean by `meaningful` here.